server-database/main_test.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestImage(t *testing.T) {
+	image := Image{"title", "altText", "url"}
+
+	AssertStrings(t, image.Title, "title")
+
+	AssertStrings(t, image.AltText, "altText")
+
+	AssertStrings(t, image.Url, "url")
+
+}
+
+func TestLoadENV(t *testing.T) {
+	closer := envSetter(map[string]string{
+		"EXECUTION_ENVIRONMENT": "test",
+	})
+	LoadENV()
+	AssertStrings(t, os.Getenv("EXECUTION_ENVIRONMENT"), "test")
+	t.Cleanup(closer)
+}
+
+func TestConnectToDB(t *testing.T) {
+	closer := envSetter(map[string]string{
+		"EXECUTION_ENVIRONMENT": "test",
+		"DB_URL":                "postgresql://postgres:postgres@localhost:5432/go_server_test_db",
+	})
+
+	LoadENV()
+	ConnectToDB()
+	defer conn.Close(context.Background())
+
+	t.Cleanup(closer)
+}
+
+func TestMain(t *testing.T) {
+
+	closer := envSetter(map[string]string{
+		"EXECUTION_ENVIRONMENT": "test",
+		"DB_URL":                "postgresql://postgres:postgres@localhost:5432/go_server_test_db",
+	})
+
+	LoadENV()
+	ConnectToDB()
+
+	t.Run("Get /", func(t *testing.T) {
+		request, _ := http.NewRequest(http.MethodGet, "/", nil)
+		response := httptest.NewRecorder()
+
+		IndexHandler(response, request)
+
+		got := response.Body.String()
+		want := "Hello World"
+
+		AssertStrings(t, got, want)
+	})
+
+	t.Run("Get /images.json", func(t *testing.T) {
+		request, _ := http.NewRequest(http.MethodGet, "/images.json", nil)
+		response := httptest.NewRecorder()
+
+		ImagesHandler(response, request)
+
+		var got []Image
+
+		decoder := json.NewDecoder(response.Body)
+		decoder.Decode(&got)
+
+		if len(got) == 0 {
```

Comment:
Let's assert on the actual output, not just that there was some.

If you're going to use an actual database for testing you should either:
1. Pre-load it with known test data
2. Start it empty, and populate test data as part of the test.

Alternatively, you can implement a fake database, but for this simple application, a real one is probably fine. In real life, you'd probably want a mix of both styles of tests - a few end-to-end test that use a real database (to make sure everything _actually_ works), and a lot of tests with a fake database which let you exercise your Go logic quickly and precisely.

Reply:
Got it, I have updated tests to check for the data in db. 

Also added setup/teadown to populate and remove data from db. 

Thanks!